src/SingleCellAnnotator.imjoy.html

```diff
@@ -106,22 +106,40 @@
   });
 };
 
-const colormap = {'Microtubule': 0, 'Protein': 1, 'ER': 0, 'Nuclei': 2, 'Mask': 3}
+const colormap = {'Microtubule': 0, 'Protein': 1, 'ER': 4, 'Nuclei': 2, 'Mask': 3}
```

Comment:
You can avoid these if check if you construct a channel array like this:
```js
const color_channels = [
 ['Microtubule', 'ER'],
 ['Protein', 'ER'],
 ['Nuclei'],
 ['Mask']
]

// replace channel name with data
for(let i=0;i<color_channels.length;i++){
  for(let j=0;j<color_channels[i].length;j++)
  {
  const ch_name = color_channels[i][j]
   color_channels[i][j] = await getImageData(channels[ch_name])
  }
}

const data = imgData.data
for(j=0;j<3;j++){
   for(i=0;i<channel_offset;i++){
         // loop over color channels, e.g. for red channel, j=0, it will fill with the sum of Microtubule and ER channel data
         for(let k=0;k<color_channels[j].length;k++){
               const src = color_channels[j][k].data;
               const index = i*4+j
               data[index] = 0;
              if(brightness===1){
                data[index] = data[index] + src[i]
              }
              else{
                v =  data[index] + src[i]*brightness;
                data[index] = v>255?255:parseInt(v);
              }
         }
   }
}

```

Reply:
I dont see where you check if er channel is turned on

Reply:
You can add an if statement to check if the corresponding channels is turned on before “getImageData”.

Reply:
I think this was implemented already